src/main/java/com/github/kagkarlsson/scheduler/task/Task.java

```diff
@@ -53,5 +63,31 @@ public String toString() {
 		return "Task " +
 				"task=" + getName();
 	}
+
+	public interface Serializer<T> {
+		byte[] serialize(T data);
+		T deserialize(byte[] serializedData);
+
+		Serializer JAVA_SERIALIZER = new Serializer<Object>() {
+			public byte[] serialize(Object data) {
```

Comment:
Ah, can't do that here, it's a static field on the interface.. Would have to create a new instance per type, but that feels a bit too much perhaps?